test/integration/utils.go

```diff
@@ -0,0 +1,350 @@
+	}
+	return cns, nil
+}
+
+func loadAzilium(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet) (appsv1.DaemonSet, error) {
+	installFlag := os.Getenv(envInstallAzilium)
+	if aziliumScenario, err := strconv.ParseBool(installFlag); err == nil && aziliumScenario {
+		log.Printf("Env %v set to true, deploy azure-ipam and cilium-cni", envInstallAzilium)
+		cns.Spec.Template.Spec.InitContainers[0].Args = []string{"deploy", "azure-ipam", "-o", "/opt/cni/bin/azure-ipam"}
+
+		// setup the CNS ciliumconfigmap
+		if err := k8sutils.MustSetupConfigMap(ctx, clientset, cnsCiliumConfigMapPath); err != nil {
+			return cns, errors.Wrap(err, "failed to setup Cilium configMap")
+		}
+	} else {
+		log.Printf("Env %v not set to true, skipping", envInstallAzilium)
+	}
+	return cns, nil
+}
+
+func loadCiliumOverlay(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet) (appsv1.DaemonSet, error) {
+	installFlag := os.Getenv(envInstallOverlay)
+	if overlayScenario, err := strconv.ParseBool(installFlag); err == nil && overlayScenario {
+		log.Printf("Env %v set to true, deploy azure-ipam and cilium-cni", envInstallOverlay)
+		cns.Spec.Template.Spec.InitContainers[0].Args = []string{"deploy", "azure-ipam", "-o", "/opt/cni/bin/azure-ipam"}
+
+		// setup the CNS cns overlay configmap
+		if err := k8sutils.MustSetupConfigMap(ctx, clientset, cnsOverlayConfigMapPath); err != nil {
+			return cns, errors.Wrap(err, "failed to setup cns overlay configMap")
+		}
+	} else {
+		log.Printf("Env %v not set to true, skipping", envInstallOverlay)
+	}
+	return cns, nil
+}
+
+func loadAzureCNIOverlay(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet) (appsv1.DaemonSet, error) {
+	installFlag := os.Getenv(envInstallAzureCNIOverlay)
+	if overlayScenario, err := strconv.ParseBool(installFlag); err == nil && overlayScenario {
+		log.Printf("Env %v set to true, deploy azure-cni and azure-cns", envInstallAzureCNIOverlay)
+		cns.Spec.Template.Spec.InitContainers[0].Args = []string{"deploy", "azure-vnet", "-o", "/opt/cni/bin/azure-vnet", "azure-vnet-telemetry", "-o", "/opt/cni/bin/azure-vnet-telemetry"}
+
+		// override the volumes and volume mounts
+		cns.Spec.Template.Spec.Volumes = volumesForAzureCNIOverlay()
+		cns.Spec.Template.Spec.InitContainers[0].VolumeMounts = dropgzVolumeMountsForAzureCNIOverlay()
+		cns.Spec.Template.Spec.Containers[0].VolumeMounts = cnsVolumeMountsForAzureCNIOverlay()
+
+		// set up the CNS configMap for azure cni overlay
+		if err := k8sutils.MustSetupConfigMap(ctx, clientset, cnsAzureCNIOverlayConfigMapPath); err != nil {
+			return cns, errors.Wrap(err, "failed to setup CNS configMap for azure cni overlay")
+		}
+	} else {
+		log.Printf("Env %v not set to true, skipping", envInstallAzureCNIOverlay)
+	}
+	return cns, nil
+}
+
+func loadDualstackOverlay(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet) (appsv1.DaemonSet, error) {
```

Comment:
The functions `loadDualstackOverlay`, `loadAzureCNIOverlay`, `loadCiliumOverlay` and `loadAzilium`. All have the same method params. `func loadCNSDaemonset(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet)`. Wondering if having a single function that provides the cns daemonset can be achieved to avoid duplicate code ? And would be easier to add another type of cns daemonset if needed. 

For example:
```
struct cnsDaemonset {
	name string
	installFlag string
	initContainerArgs []string
	volumes []corev1.Volume
	initContainerVolumeMounts []corev1.VolumeMount
	containerVolumeMounts []corev1.VolumeMount
	configMapPath string
}

var cnsDaemonsetMap = map[string]cnsDaemonset{
	envInstallAzureCNIOverlay: cnsDaemonset{
		installFlag: envInstallAzureCNIOverlay,
		initContainerArgs: []string{
			"deploy", "azure-vnet", "-o", "/opt/cni/bin/azure-vnet", "azure-vnet-telemetry", "-o", "/opt/cni/bin/azure-vnet-telemetry"}
		volumes: volumesForAzureCNIOverlay(),
		initContainerVolumeMounts: dropgzVolumeMountsForAzureCNIOverlay(),
		containerVolumeMounts: cnsVolumeMountsForAzureCNIOverlay(),
		configMapPath: cnsSwiftConfigMapPath,
	},
}

func setupCNSDaemonset(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet, flag string) (appsv1.DaemonSet, error) {
	installFlag = os.Getenv(flag)

	if scenario, err := strconv.ParseBool(cnsDaemonsetMap[installFlag]); err == nil && scenario {
		log.Printf("Env %v set to true, deploy %s", scenario, cnsDaemonsetMap[flag].name)
		cns.Spec.Template.Spec.InitContainers[0].Args = cnsDaemonsetMap[installFlag].initContainerArgs
		if err := k8sutils.MustSetupConfigMap(ctx, clientset, cnsDaemonsetMap[installFlag].configMapPath); err != nil {
			return cns, errors.Wrap(err, "failed to setup CNS %s configMap", cnsDaemonsetMap[flag].name)
		}
	} else {
		log.Printf("Env %v not set to true, skipping", cnsDaemonsetMap[flag])
	}
	return cns, nil
}

```

Reply:
Done